python/lsst/ts/wep/task/calcZernikesNeuralTask.py

```diff
@@ -0,0 +1,1964 @@
+            "Creating Zernike table: intra stamps=%d, extra stamps=%d",
+            len(intraStamps),
+            len(extraStamps),
+        )
+
+        # Add average row
+        row_data = {
+            "label": "average",
+            "used": True,
+            "intra_field_x": np.nan,
+            "intra_field_y": np.nan,
+            "extra_field_x": np.nan,
+            "extra_field_y": np.nan,
+            "intra_centroid_x": np.nan,
+            "intra_centroid_y": np.nan,
+            "extra_centroid_x": np.nan,
+            "extra_centroid_y": np.nan,
+            "intra_mag": np.nan,
+            "extra_mag": np.nan,
+            "intra_sn": np.nan,
+            "extra_sn": np.nan,
+            "intra_entropy": np.nan,
+            "extra_entropy": np.nan,
+            "intra_frac_bad_pix": np.nan,
+            "extra_frac_bad_pix": np.nan,
+            "intra_max_power_grad": np.nan,
+            "extra_max_power_grad": np.nan,
+            "intra_fx": np.nan,
+            "extra_fx": np.nan,
+            "intra_fy": np.nan,
+            "extra_fy": np.nan,
+        }
+
+        # Add Zernike coefficients
+        for i, j in enumerate(self.config.nollIndices):
+            if i < len(zkCoeffAvg):
+                row_data[f"Z{j}"] = zkCoeffAvg[i] * u.um
+            else:
+                row_data[f"Z{j}"] = 0.0 * u.um
+
+        zkTable.add_row(row_data)
+        self.log.debug("Added average row with %d Zernike terms", len(self.config.nollIndices))
+
+        # Add individual donut rows only if we have donut stamps
+        # For TARTS, we typically have one set of coefficients per exposure
+        # We'll create one row per donut if available, otherwise just the
+        # average row
+        max_stamps = max(len(extraStamps), len(intraStamps))
+        self.log.debug(
+            "Creating Zernike table: %d extra stamps, %d intra stamps, "
+            "zkCoeffRaw shape: %s",
+            len(extraStamps), len(intraStamps), zkCoeffRaw.shape
+        )
+
+        # Add individual donut rows
+        for i, (intraStamp, extraStamp) in enumerate(zip_longest(intraStamps, extraStamps)):
+            # Get the zernike coefficients for this donut
+            # For TARTS, we have individual coefficients per donut in
+            # total_zernikes
+            if zkCoeffRaw.ndim == 2 and zkCoeffRaw.shape[0] > i:
+                zk = zkCoeffRaw[i]
+                self.log.debug("Using individual coefficients for donut %d", i+1)
+            elif zkCoeffRaw.ndim == 2 and zkCoeffRaw.shape[0] == 1:
+                # Single set of coefficients for all donuts (fallback)
+                zk = zkCoeffRaw[0]
+                if i == 0:  # Only log once
+                    self.log.debug(
+                        "Using single coefficient set for all %d donuts (fallback)",
+                        max_stamps,
+                    )
+            else:
+                # If we don't have individual coefficients, use the average
```

Comment:
Again, when would this be the case though? what in tarts makes it possible to only report the average and not the pair-by-pair estimate as well?